Mathlib/RingTheory/HopfAlgebra.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Ali Ramsey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ali Ramsey
+-/
+import Mathlib.RingTheory.Bialgebra
+
+/-!
+# Hopf algebras
+
+In this file we define `HopfAlgebra`, and provide instances for:
+
+* Commutative semirings: `CommSemiring.toHopfAlgebra`
+
+# Main definitions
+
+* `HopfAlgebra R A` : the Hopf algebra structure on an `R`-bialgebra `A`.
+* `HopfAlgebra.antipode R A` : The `R`-linear map `A →ₗ[R] A`
+
+## TODO
+
+* Uniqueness of Hopf algebra structure on a bialgebra (i.e. if the algebra and coalgebra structures
+agree then the antipodes must also agree).
+
+* S(1)=1 and S(ab)=S(b)S(a), so in particular if A is commutative then S is an algebra homomorphism.
+
+* If H is commutative then S is necessarily a bijection (and I guess its square must be the
+identity?)
+
+References for these facts: Christian Kassel "Quantum Groups" (Springer GTM), around Prop III.3.1,
+Theorem III.3.4 etc.
```

Comment:
can you add this to `references.bib`?

Reply:
I guess you can just copy

https://github.com/leanprover-community/mathlib4/blob/65037e3470a82737b4ce82d6a0c4e4f7534fbf47/docs/references.bib#L1812-L1825

with a different chapter number

Reply:
You already added it, but only chapter 2:
```text 
@InBook{          kasselTensorProducts1995,
  title         = {Tensor {{Products}}},
  booktitle     = {Quantum {{Groups}}},
  author        = {Kassel, Christian},
  year          = {1995},
  volume        = {155},
  pages         = {23--38},
  publisher     = {{Springer New York}},
  address       = {{New York, NY}},
  doi           = {10.1007/978-1-4612-0783-2_2},
  urldate       = {2023-09-28},
  collaborator  = {Kassel, Christian},
  isbn          = {978-1-4612-6900-7 978-1-4612-0783-2}
}
```
Should I delete this, replace with a reference to the entire book, and edit your reference to it in RingTheory.TensorProduct, in this PR?

Reply:
That sounds fine to me too.